src/main/java/io/odpf/depot/http/record/HttpRequestRecord.java

```diff
@@ -10,12 +10,13 @@
 import org.apache.http.util.EntityUtils;
 
 import java.io.IOException;
+import java.util.List;
 
 @AllArgsConstructor
 @Getter
 public class HttpRequestRecord {
 
-    private final Long index;
+    private final List<Integer> index;
```

Comment:
Let's change it to Set<Integer> as there is no ordering here. 
Set<Integer> indexes = new HasSet<>();
Also add methods to add in the set. Instead of constructor. 
remove the @AllArgsConstructor and @Getter. Check which fields are needed for getter. Take only those.
Maybe you can have an iterator() for the indexes rather than returning the whole set.